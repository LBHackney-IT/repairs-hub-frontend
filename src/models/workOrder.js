import { formatISO, isSameDay } from 'date-fns'
import {
  HIGH_PRIORITY_CODES,
  PRIORITY_CODES_REQUIRING_APPOINTMENTS,
} from '@/utils/helpers/priorities'

import {
  CLOSED_STATUS_DESCRIPTIONS,
  CLOSED_STATUS_DESCRIPTIONS_FOR_OPERATIVES,
} from '@/utils/statusCodes'

export class WorkOrder {
  constructor(workOrderData) {
    Object.assign(this, workOrderData)
  }

  isHigherPriority = () => {
    return HIGH_PRIORITY_CODES.includes(this.priorityCode)
  }

  isAppointmentRequired = () => {
    return PRIORITY_CODES_REQUIRING_APPOINTMENTS.includes(this.priorityCode)
  }

  canBeScheduled = () => {
    return (
      this.statusAllowsScheduling() &&
      this.isAppointmentRequired() &&
      !this.isOutOfHoursGas()
    )
  }

  isOutOfHoursGas = () => {
    const GasBreakdownContractorReference = 'H04'
    const OohTradeCode = 'OO'

    var contractorReference = this.contractorReference

    if (contractorReference != GasBreakdownContractorReference) return false // contractor must be "H04"
    var tradeCode = this.tradeCode
    return tradeCode == OohTradeCode
  }

  completionReason = () => {
    return this.status === 'Work Completed' ? 'Completed' : this.status
  }

  appointmentISODatePassed = () => {
    if (
      !this.appointment ||
      !this.appointment.date ||
      !this.appointment.start
    ) {
      return false
    }

    const currentISODate = formatISO(new Date(), { representation: 'date' })

    const appointmentISODate = formatISO(
      new Date(`${this.appointment.date}T${this.appointment.start}`),
      { representation: 'date' }
    )

    return currentISODate >= appointmentISODate
  }

  appointmentIsToday = () => {
    if (
      !this.appointment ||
      !this.appointment.date ||
      !this.appointment.start
    ) {
      return false
    } else {
      return isSameDay(
        new Date(),
        new Date(`${this.appointment.date}T${this.appointment.start}`)
      )
    }
  }

  statusAllowsScheduling = () => {
    return !CLOSED_STATUS_DESCRIPTIONS.includes(this.status)
  }

  targetTimePassed = () => {
    const currentTime = new Date().getTime()
    const targetTime = new Date(`${this.target}`).getTime()

    return currentTime > targetTime
  }

  hasBeenVisited = () =>
    CLOSED_STATUS_DESCRIPTIONS_FOR_OPERATIVES.includes(this.status)
}
