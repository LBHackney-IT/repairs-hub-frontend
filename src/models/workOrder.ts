import { formatISO, isSameDay } from 'date-fns'
import {
  HIGH_PRIORITY_CODES,
  PRIORITY_CODES_REQUIRING_APPOINTMENTS,
} from '@/utils/helpers/priorities'

import {
  CLOSED_STATUS_DESCRIPTIONS,
  CLOSED_STATUS_DESCRIPTIONS_FOR_OPERATIVES,
} from '@/utils/statusCodes'
import { Operative } from './Operative'
import { Appointment } from './appointmentModel'

export class WorkOrder {
  reference: string
  description: string
  externalAppointmentManagementUrl: string
  startTime: string
  appointment: Appointment
  status: string
  dateRaised: string
  tradeDescription: string
  isFollowOn: boolean
  contractorReference: string
  tradeCode: string
  priorityCode
  target
  budgetCode
  closedDated
  operatives: Operative[]
  owner
  priority: string
  raisedBy: string
  callerName: string
  callerNumber: string

  constructor(workOrderData) {
    Object.assign(this, workOrderData)
  }

  isHigherPriority = () => {
    return HIGH_PRIORITY_CODES.includes(this.priorityCode)
  }

  isAppointmentRequired = () => {
    return PRIORITY_CODES_REQUIRING_APPOINTMENTS.includes(this.priorityCode)
  }

  canBeScheduled = () => {
    return (
      this.statusAllowsScheduling() &&
      this.isAppointmentRequired() &&
      !this.isOutOfHoursGas()
    )
  }

  isOutOfHoursGas = () => {
    const gasBreakdownContractorReference = 'H04'
    const oohTradeCode = 'OO'

    const contractorReference = this.contractorReference

    if (contractorReference != gasBreakdownContractorReference) return false // contractor must be "H04"
    const tradeCode = this.tradeCode
    return tradeCode == oohTradeCode
  }

  completionReason = () => {
    return this.status === 'Work Completed' ? 'Completed' : this.status
  }

  appointmentISODatePassed = () => {
    if (
      !this.appointment ||
      !this.appointment.date ||
      !this.appointment.start
    ) {
      return false
    }

    const currentISODate = formatISO(new Date(), { representation: 'date' })

    const appointmentISODate = formatISO(
      new Date(`${this.appointment.date}T${this.appointment.start}`),
      { representation: 'date' }
    )

    return currentISODate >= appointmentISODate
  }

  appointmentIsToday = () => {
    if (
      !this.appointment ||
      !this.appointment.date ||
      !this.appointment.start
    ) {
      return false
    } else {
      return isSameDay(
        new Date(),
        new Date(`${this.appointment.date}T${this.appointment.start}`)
      )
    }
  }

  statusAllowsScheduling = () => {
    return !CLOSED_STATUS_DESCRIPTIONS.includes(this.status)
  }

  targetTimePassed = () => {
    const currentTime = new Date().getTime()
    const targetTime = new Date(`${this.target}`).getTime()

    return currentTime > targetTime
  }

  hasBeenVisited = () =>
    CLOSED_STATUS_DESCRIPTIONS_FOR_OPERATIVES.includes(this.status)
}
