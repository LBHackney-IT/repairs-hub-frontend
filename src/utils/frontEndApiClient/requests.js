import axios from 'axios'
import { paramsSerializer } from '@/utils/urls'

export const frontEndApiRequest = async ({
  method,
  path,
  params,
  requestData,
  paramsSerializer,
}) => {
  const { data } = await axios({
    method: method,
    url: path,
    params: params,
    headers: {
      // 'content-type': 'multipart/form-data',
      //   'Content-Type': 'application/json',
    },
    ...(requestData && { data: requestData }),
    ...(paramsSerializer && { paramsSerializer }),
  })

  return data
}

export const fetchFeatureToggles = async () => {
  try {
    const configurationData = await frontEndApiRequest({
      method: 'GET',
      path: '/api/toggles',
    })

    return configurationData?.[0]?.featureToggles || {}
  } catch (e) {
    console.error(
      `Error fetching toggles from configuration API: ${JSON.stringify(e)}`
    )

    return {}
  }
}

export const createSorExistenceValidator = (
  tradeCode,
  propertyRef,
  contractorRef,
  isRaisable
) => {
  return async (codesForValidation) => {
    const validationResults = {
      allCodesValid: false,
      validCodes: [],
      invalidCodes: [],
    }

    if (codesForValidation.length === 0) {
      return { ...validationResults, allCodesValid: true }
    }

    try {
      const sorCodes = await frontEndApiRequest({
        method: 'get',
        path: '/api/schedule-of-rates/check',
        params: {
          tradeCode: tradeCode,
          propertyReference: propertyRef,
          contractorReference: contractorRef,
          sorCode: codesForValidation,
          isRaisable: isRaisable,
        },
        ...(paramsSerializer && { paramsSerializer }),
      })

      validationResults.validCodes = sorCodes.filter((sorCodeObj) =>
        codesForValidation.includes(sorCodeObj.code)
      )

      const validCodes = validationResults.validCodes.map(
        (sorCodeObj) => sorCodeObj.code
      )

      validationResults.invalidCodes = codesForValidation.filter(
        (code) => !validCodes.includes(code)
      )

      if (validationResults.validCodes.length === codesForValidation.length) {
        validationResults.allCodesValid = true
      }

      return validationResults
    } catch (e) {
      throw new Error(
        `Cannot fetch SOR codes for validation: ${JSON.stringify(e)}`
      )
    }
  }
}
