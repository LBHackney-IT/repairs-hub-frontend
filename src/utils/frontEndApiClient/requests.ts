import axios, { AxiosRequestConfig, Method } from 'axios'
import { paramsSerializer } from '@/utils/urls'

export const frontEndApiRequest = async ({
  method,
  path,
  params = null,
  requestData,
  paramsSerializer,
  timeout,
}: {
  method: Method
  path: string
  params?: object
  requestData?: object
  paramsSerializer?: any
  timeout?: number
}) => {
  const config: AxiosRequestConfig = {
    method: method,
    url: path,
    timeout,
    ...(requestData && { data: requestData }),
    ...(paramsSerializer && { paramsSerializer }),
  }

  if (params) {
    config.params = params
  }

  const { data } = await axios(config)

  return data
}

export const fetchSimpleFeatureToggles = async () => {
  const featureToggleData = await frontEndApiRequest({
    method: 'get',
    path: '/api/simple-feature-toggle',
  })

  return featureToggleData
}

export const fetchFeatureToggles = async () => {
  try {
    const configurationData = await frontEndApiRequest({
      method: 'GET',
      path: '/api/toggles',
    })

    return configurationData?.[0]?.featureToggles || {}
  } catch (e) {
    console.error(
      `Error fetching toggles from configuration API: ${JSON.stringify(e)}`
    )

    return {}
  }
}

export const createSorExistenceValidator = (
  tradeCode,
  propertyRef,
  contractorRef,
  isRaisable
) => {
  return async (codesForValidation) => {
    const validationResults = {
      allCodesValid: false,
      validCodes: [],
      invalidCodes: [],
    }

    if (codesForValidation.length === 0) {
      return { ...validationResults, allCodesValid: true }
    }

    try {
      const sorCodes = await frontEndApiRequest({
        method: 'get',
        path: '/api/schedule-of-rates/check',
        params: {
          tradeCode: tradeCode,
          propertyReference: propertyRef,
          contractorReference: contractorRef,
          sorCode: codesForValidation,
          isRaisable: isRaisable,
        },
        ...(paramsSerializer && { paramsSerializer }),
      })

      validationResults.validCodes = sorCodes.filter((sorCodeObj) =>
        codesForValidation.includes(sorCodeObj.code)
      )

      const validCodes = validationResults.validCodes.map(
        (sorCodeObj) => sorCodeObj.code
      )

      validationResults.invalidCodes = codesForValidation.filter(
        (code) => !validCodes.includes(code)
      )

      if (validationResults.validCodes.length === codesForValidation.length) {
        validationResults.allCodesValid = true
      }

      return validationResults
    } catch (e) {
      throw new Error(
        `Cannot fetch SOR codes for validation: ${JSON.stringify(e)}`
      )
    }
  }
}
