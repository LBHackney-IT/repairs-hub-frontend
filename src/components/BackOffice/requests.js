import { frontEndApiRequest } from '../../utils/frontEndApiClient/requests'

export const fetchContractors = async () => {
  return await frontEndApiRequest({
    method: 'get',
    path: '/api/contractors?getAllContractors=true',
  })
}

export const fetchActiveContractsByContractorReference = async (
  isActive = true,
  contractorReference
) => {
  return await frontEndApiRequest({
    method: 'get',
    path: `/api/backoffice/contracts?isActive=${isActive}true&ContractorReference=${contractorReference}`,
  })
}
