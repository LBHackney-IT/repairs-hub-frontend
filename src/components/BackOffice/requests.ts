import { frontEndApiRequest } from '../../utils/frontEndApiClient/requests'
import { APIResponseError, ApiResponseType } from '../../types/requests/types'
import Contract from '@/models/contract'

export const fetchContractors = async () => {
  return await frontEndApiRequest({
    method: 'get',
    path: '/api/contractors?getAllContractors=true',
  })
}

export const fetchContracts = async (
  isActive?: boolean,
  contractorReference?: string
): Promise<ApiResponseType<Contract[] | null>> => {
  try {
    const contracts = await frontEndApiRequest({
      method: 'get',
      path: `/api/backoffice/contracts?isActive=${
        isActive ? isActive : ''
      }&contractorReference=${contractorReference ? contractorReference : ''}`,
    })

    return {
      success: true,
      response: contracts,
      error: null,
    }
  } catch (e) {
    console.error('An error has occurred:', e.response)

    return {
      success: false,
      response: null,
      error: new APIResponseError(
        e.response?.status === 404
          ? `Could not find any contracts`
          : `Oops, an error occurred: ${
              e.response?.status
            } with message: ${JSON.stringify(e.response?.data?.message)}`
      ),
    }
  }
}

export const fetchContract = async (
  contractReference: string
): Promise<ApiResponseType<Contract | null>> => {
  try {
    const contract = await frontEndApiRequest({
      method: 'get',
      path: `/api/backoffice/contract/${contractReference}`,
    })

    return {
      success: true,
      response: contract,
      error: null,
    }
  } catch (e) {
    console.error('An error has occurred:', e.response)

    return {
      success: false,
      response: null,
      error: new APIResponseError(
        e.response?.status === 404
          ? `Could not find contract with contract reference ${contractReference}`
          : `Oops, an error occurred: ${
              e.response?.status
            } with message: ${JSON.stringify(e.response?.data?.message)}`
      ),
    }
  }
}
