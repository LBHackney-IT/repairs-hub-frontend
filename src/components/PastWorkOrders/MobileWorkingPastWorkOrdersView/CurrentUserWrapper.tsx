import { useState, useEffect } from 'react'
import { frontEndApiRequest } from '@/utils/frontEndApiClient/requests'
import Spinner from '../../Spinner'
import ErrorMessage from '../../Errors/ErrorMessage'
import MobileWorkingPastWorkOrdersView from './MobileWorkingPastWorkOrdersView'

interface CurrentUser {
  sub: string
  name: string
  email: string
  varyLimit: string
  raiseLimit: string
  contractors: any[] // Adjust!
  operativePayrollNumber: string | null
  isOneJobAtATime: boolean
}

const CurrentUserWrapper = () => {
  const [currentUser, setCurrentUser] = useState<CurrentUser | null>(null)
  const [error, setError] = useState<string | null>()
  const [loading, setLoading] = useState(false)

  const getOperativeWorkOrderView = async () => {
    setLoading(true)
    setError(null)

    try {
      const currentUser = await frontEndApiRequest({
        method: 'get',
        path: '/api/hub-user',
      })
      console.log({ currentUser })
      currentUser.isOneJobAtATime = true
      currentUser.operativePayrollNumber = '016062'
      setCurrentUser(currentUser)
    } catch (e) {
      setError(
        `Oops an error occurred with error status: ${e.response?.status} with message: ${e.response?.data?.message}`
      )
    }

    setLoading(false)
  }

  useEffect(() => {
    getOperativeWorkOrderView()
  }, [])

  return (
    <>
      {loading ? (
        <Spinner />
      ) : (
        <>
          {!!currentUser && (
            <MobileWorkingPastWorkOrdersView currentUser={currentUser} />
          )}
          {error && <ErrorMessage label={error} />}
        </>
      )}
    </>
  )
}

export default CurrentUserWrapper
