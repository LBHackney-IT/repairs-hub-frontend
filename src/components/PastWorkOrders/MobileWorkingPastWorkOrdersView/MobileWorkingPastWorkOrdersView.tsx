import { useState, useEffect, useRef } from 'react'
import { frontEndApiRequest } from '@/utils/frontEndApiClient/requests'
import { beginningOfDay, daysBeforeDateRangeExcWeekend } from '@/utils/time'
import { longMonthWeekday } from '@/utils/date'

import { Select } from '../../Form'
import DatePicker from '../../DatePicker/Index'
import Spinner from '../../Spinner'
import ErrorMessage from '../../Errors/ErrorMessage'
import WarningInfoBox from '../../Template/WarningInfoBox'
import Meta from '../../Meta'
import { WorkOrder } from '../../../models/workOrder'
import { MobileWorkingPastWorkOrderListItems } from './MobileWorkingPastWorkOrderListItems'
import {
  CurrentUser,
  WorkOrderType,
  WorkOrdersType,
} from '@/root/src/types/variations/types'

const SIXTY_SECONDS = 60 * 1000

const MobileWorkingPastWorkOrdersView = ({ currentUser }) => {
  const currentDate = beginningOfDay(new Date())
  const yesterday = new Date()
  yesterday.setDate(currentDate.getDate() - 1)
  const lastSevenDays = daysBeforeDateRangeExcWeekend(currentDate, 7)
  const [visitedWorkOrders, setVisitedWorkOrders] = useState(null)
  const [sortedWorkOrders, setSortedWorkOrders] = useState(null)
  const [error, setError] = useState<string | null>()
  const [selectedDate, setSelectedDate] = useState<Date>(yesterday)
  const targetDate = selectedDate.toISOString().split('T')[0]

  const getOperativeWorkOrderView = async () => {
    setError(null)

    try {
      const data = await frontEndApiRequest({
        method: 'get',
        path: `/api/operatives/017233/workOrdersNew?date=${targetDate}`,
      })

      const workOrders: WorkOrdersType = data.map(
        (wo: WorkOrderType) => new WorkOrder(wo)
      )

      const visitedWorkOrders = workOrders.filter((wo: WorkOrderType) =>
        wo.hasBeenVisited()
      )

      const sortedWorkOrderItems = sortWorkOrderItems(currentUser, workOrders)
      setVisitedWorkOrders(visitedWorkOrders)
      setSortedWorkOrders(sortedWorkOrderItems)
    } catch (e) {
      setVisitedWorkOrders(null)
      setSortedWorkOrders(null)

      console.error('An error has occured:', e)
      setError(
        `Oops an error occurred with error status: ${e.response?.status} with message: ${e.response?.data?.message}`
      )
    }
  }

  const intervalRef = useRef(null)

  useEffect(() => {
    // dont try fetch if currentUser not loaded yet
    if (currentUser === null) return

    // initial fetch (otherwise it wont fetch until interval)
    getOperativeWorkOrderView()

    const intervalId = setInterval(() => {
      getOperativeWorkOrderView()
    }, SIXTY_SECONDS)

    intervalRef.current = intervalId

    return () => {
      clearInterval(intervalRef.current)
    }
  }, [currentUser?.operativePayrollNumber, targetDate])

  const sortWorkOrderItems = (
    currentUser: CurrentUser,
    workOrders: WorkOrdersType
  ) => {
    const inProgressWorkOrders = workOrders.filter((wo) => !wo.hasBeenVisited())

    const startedWorkOrders = workOrders.filter(
      (wo) => !wo.hasBeenVisited() && !!wo.appointment.startedAt?.length
    )

    // The operative is NOT an OJAAT operative
    if (!currentUser.isOneJobAtATime) return inProgressWorkOrders

    // If the operative has started a work order
    if (startedWorkOrders?.length) return startedWorkOrders

    // Return the next unstarted work order
    return inProgressWorkOrders
      .sort((a, b) => {
        return a.appointment.assignedStart.localeCompare(
          b.appointment.assignedStart
        )
      })
      .slice(0, 1)
  }

  const handleDateChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
    const selectedDate = event.target.value
    setSelectedDate(new Date(selectedDate))
  }

  return (
    <>
      <Meta title="Manage past work orders" />
      <div className="mobile-work-order-container">
        <DatePicker options={lastSevenDays} handleChange={handleDateChange} />
        {sortedWorkOrders === null ? (
          <Spinner />
        ) : (
          <>
            {sortedWorkOrders?.length || visitedWorkOrders?.length ? (
              <ol className="lbh-list mobile-working-work-order-list">
                <MobileWorkingPastWorkOrderListItems
                  workOrders={[...sortedWorkOrders, ...visitedWorkOrders]}
                  currentUser={currentUser}
                />
              </ol>
            ) : (
              <WarningInfoBox
                header="No work orders displayed"
                text="Please contact your supervisor"
              />
            )}
            {error && <ErrorMessage label={error} />}
          </>
        )}
      </div>
    </>
  )
}

export default MobileWorkingPastWorkOrdersView
