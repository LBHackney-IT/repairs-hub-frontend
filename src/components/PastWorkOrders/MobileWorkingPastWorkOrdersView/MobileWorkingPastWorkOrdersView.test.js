import React from 'react'
import { render, screen, fireEvent } from '@testing-library/react'
import MobileWorkingPastWorkOrdersView from './MobileWorkingPastWorkOrdersView'
import { frontEndApiRequest } from '@/utils/frontEndApiClient/requests'
import { WorkOrder } from '../../../models/workOrder'

// Mock dependencies
jest.mock('@/utils/frontEndApiClient/requests')
jest.mock('../../Spinner', () => jest.fn(() => <div>Loading...</div>))

const mockCurrentUser = {
  sub: '112088470639703810484',
  name: 'Alexander Perez-Davies',
  email: 'alexander.perez-davies@hackney.gov.uk',
  varyLimit: '10000',
  raiseLimit: '10000',
  contractors: [],
  operativePayrollNumber: null,
  isOneJobAtATime: false,
}

describe('MobileWorkingPastWorkOrdersView Component', () => {
  beforeEach(() => {
    jest.clearAllMocks()
  })

  it('should match snapshot when loading', async () => {
    // Mock the API response for a successful fetch
    frontEndApiRequest.mockResolvedValueOnce([
      new WorkOrder({
        id: 1,
        appointment: { assignedStart: '2024-11-07T09:00' },
      }),
      new WorkOrder({
        id: 2,
        appointment: { assignedStart: '2024-11-07T10:00' },
      }),
    ])

    const { asFragment } = render(
      <MobileWorkingPastWorkOrdersView currentUser={mockCurrentUser} />
    )

    // Assert that the component matches the snapshot
    expect(asFragment()).toMatchSnapshot()
  })

  it('should match snapshot when there is no work orders', async () => {
    // Mock the API response for an empty list
    frontEndApiRequest.mockResolvedValueOnce([])

    const { asFragment } = render(
      <MobileWorkingPastWorkOrdersView currentUser={mockCurrentUser} />
    )

    // Wait for the component to re-render after API request
    await screen.findByText('No work orders displayed')

    // Assert that the component matches the snapshot
    expect(asFragment()).toMatchSnapshot()
  })

  it('should show loading spinner initially', () => {
    // Render the component with no data yet (loading state)
    const { asFragment } = render(
      <MobileWorkingPastWorkOrdersView currentUser={mockCurrentUser} />
    )

    // Assert that the component shows the loading spinner initially
    expect(screen.getByText('Loading...')).toBeInTheDocument()

    // Match snapshot for loading state
    expect(asFragment()).toMatchSnapshot()
  })

  it('should update the selected date correctly', async () => {
    const { asFragment } = render(
      <MobileWorkingPastWorkOrdersView currentUser={mockCurrentUser} />
    )

    // Select a new date
    const datePicker = screen.getByLabelText('Select date')
    const options = datePicker.querySelectorAll('option')
    options[1].selected = true // Select the second option

    // Fire change event
    fireEvent.change(datePicker, { target: { value: options[1].value } })

    // Match snapshot after date change
    expect(asFragment()).toMatchSnapshot()
  })
})
