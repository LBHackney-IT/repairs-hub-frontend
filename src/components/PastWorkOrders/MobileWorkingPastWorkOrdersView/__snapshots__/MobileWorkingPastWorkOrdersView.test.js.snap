// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileWorkingPastWorkOrdersView Component should match snapshot when loading 1`] = `
<DocumentFragment>
  <div
    class="mobile-work-order-container"
  >
    <div
      class="date-picker-container"
    >
      <label
        class="lbh-heading-h2"
        for="date-picker"
      >
        Select date
      </label>
      <select
        id="date-picker"
        name="date-picker"
      >
        <option
          value="Wed Nov 06 2024 00:00:00 GMT+0000 (Greenwich Mean Time)"
        >
           Nov 06
        </option>
        <option
          value="Tue Nov 05 2024 00:00:00 GMT+0000 (Greenwich Mean Time)"
        >
           Nov 05
        </option>
        <option
          value="Mon Nov 04 2024 00:00:00 GMT+0000 (Greenwich Mean Time)"
        >
           Nov 04
        </option>
        <option
          value="Fri Nov 01 2024 00:00:00 GMT+0000 (Greenwich Mean Time)"
        >
           Nov 01
        </option>
        <option
          value="Thu Oct 31 2024 00:00:00 GMT+0000 (Greenwich Mean Time)"
        >
           Oct 31
        </option>
      </select>
    </div>
    <div>
      Loading...
    </div>
  </div>
</DocumentFragment>
`;

exports[`MobileWorkingPastWorkOrdersView Component should match snapshot when there is no work orders 1`] = `
<DocumentFragment>
  <div
    class="mobile-work-order-container"
  >
    <div
      class="date-picker-container"
    >
      <label
        class="lbh-heading-h2"
        for="date-picker"
      >
        Select date
      </label>
      <select
        id="date-picker"
        name="date-picker"
      >
        <option
          value="Wed Nov 06 2024 00:00:00 GMT+0000 (Greenwich Mean Time)"
        >
           Nov 06
        </option>
        <option
          value="Tue Nov 05 2024 00:00:00 GMT+0000 (Greenwich Mean Time)"
        >
           Nov 05
        </option>
        <option
          value="Mon Nov 04 2024 00:00:00 GMT+0000 (Greenwich Mean Time)"
        >
           Nov 04
        </option>
        <option
          value="Fri Nov 01 2024 00:00:00 GMT+0000 (Greenwich Mean Time)"
        >
           Nov 01
        </option>
        <option
          value="Thu Oct 31 2024 00:00:00 GMT+0000 (Greenwich Mean Time)"
        >
           Oct 31
        </option>
      </select>
    </div>
    <div
      class="warning-info-box govuk-inset-text lbh-inset-text"
    >
      <div
        class="lbh-warning-text govuk-warning-text"
      >
        <span
          aria-hidden="true"
          class="govuk-warning-text__icon"
        >
          !
        </span>
        <div
          class="govuk-warning-text__text"
        >
          <span
            class="govuk-warning-text__assistive"
          >
            Warning
          </span>
          <p
            class="govuk-!-margin-top-0 lbh-body-s lbh-!-font-weight-bold"
          >
            No work orders displayed
          </p>
          <p
            class="lbh-body-xs govuk-!-margin-top-1"
          >
            Please contact your supervisor
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`MobileWorkingPastWorkOrdersView Component should show loading spinner initially 1`] = `
<DocumentFragment>
  <div
    class="mobile-work-order-container"
  >
    <div
      class="date-picker-container"
    >
      <label
        class="lbh-heading-h2"
        for="date-picker"
      >
        Select date
      </label>
      <select
        id="date-picker"
        name="date-picker"
      >
        <option
          value="Wed Nov 06 2024 00:00:00 GMT+0000 (Greenwich Mean Time)"
        >
           Nov 06
        </option>
        <option
          value="Tue Nov 05 2024 00:00:00 GMT+0000 (Greenwich Mean Time)"
        >
           Nov 05
        </option>
        <option
          value="Mon Nov 04 2024 00:00:00 GMT+0000 (Greenwich Mean Time)"
        >
           Nov 04
        </option>
        <option
          value="Fri Nov 01 2024 00:00:00 GMT+0000 (Greenwich Mean Time)"
        >
           Nov 01
        </option>
        <option
          value="Thu Oct 31 2024 00:00:00 GMT+0000 (Greenwich Mean Time)"
        >
           Oct 31
        </option>
      </select>
    </div>
    <div>
      Loading...
    </div>
  </div>
</DocumentFragment>
`;

exports[`MobileWorkingPastWorkOrdersView Component should update the selected date correctly 1`] = `
<DocumentFragment>
  <div
    class="mobile-work-order-container"
  >
    <div
      class="date-picker-container"
    >
      <label
        class="lbh-heading-h2"
        for="date-picker"
      >
        Select date
      </label>
      <select
        id="date-picker"
        name="date-picker"
      >
        <option
          value="Wed Nov 06 2024 00:00:00 GMT+0000 (Greenwich Mean Time)"
        >
           Nov 06
        </option>
        <option
          value="Tue Nov 05 2024 00:00:00 GMT+0000 (Greenwich Mean Time)"
        >
           Nov 05
        </option>
        <option
          value="Mon Nov 04 2024 00:00:00 GMT+0000 (Greenwich Mean Time)"
        >
           Nov 04
        </option>
        <option
          value="Fri Nov 01 2024 00:00:00 GMT+0000 (Greenwich Mean Time)"
        >
           Nov 01
        </option>
        <option
          value="Thu Oct 31 2024 00:00:00 GMT+0000 (Greenwich Mean Time)"
        >
           Oct 31
        </option>
      </select>
    </div>
    <div>
      Loading...
    </div>
  </div>
</DocumentFragment>
`;
